extend/generic-writer/configuration/configuration.md

```diff
@@ -0,0 +1,718 @@
+```
+
+##### Column datatypes
+
+
+Optional configuration of column types. This version supports nesting (three levels) and three datatypes:
+
+- `bool` -  Boolean value  case-insensitive conversion: `t`, `true`, `yes`, `1`,`"1"` to `True` and `f`, `false`, `no` to `False`
+- `string` - String
+- `number` - Number
+- `object` - Object - valid JSON array or JSON object, e.g. ["1","2"], {"key":"val"}
+
+###### Autodetect
+
+[OPTIONAL] Default value `true
+`
+Set this option to `true` to make the parser automatically detect the above datatypes. It may be used in combination with 
+`datatype_override` option to force datatype to some columns.
+
+###### Column datatype override
+
+[OPTIONAL]
+
+The `autodetect` option in most cases takes care of the datatype conversion properly. 
+But there are some scenarios where you want make sure that the datatype conversion is forced. 
+E.g. for `phone_number` column to be treated as String a mapping should be defined as `"phone_number":"string"`.
+
+Below are options that can be used as a datatype values:
+
+if you want the value to be always a string, use `string`, 
+if you want the value to be numeric, use `number`. 
+If you want it to be Boolean, use `bool` 
+(case-insensitive conversion: `t`, `true`, `yes` to `True` and `f`, `false`, `no` to `False`)
+If the value should be an array or object `object` -  valid JSON array or JSON object, e.g. ["1","2"], {"key":"val"}
+
+**Note** If the `autodetect` option is turned off all unspecified column will be treated as a string.
+
+```json
+"request_options": {
+            ...,
+            "content": {
+                "content_type": "JSON",
+                "json_mapping": {
+                    "nesting_delimiter": "_",
+                    "chunk_size": 1,
+                    "column_data_types": {
+                        "autodetect": true,
+                        "datatype_override": [{
+                                "column": "phone",
+                                "type": "string"
+                            }, {
+                                "column": "rank",
+                                "type": "number"
+                            }, {
+                                "column": "is_active",
+                                "type": "bool"
+                            }
+                        ]
+                    }
+```
+
+See [example 007](https://bitbucket.org/kds_consulting_team/kds-team.wr-generic/src/master/docs/examples/007-simple-json-force-datatype/)
+
+##### Request Data Wrapper
+
+[OPTIONAL]
+
+A wrapper/mask of the parsed data. It needs to be json-encoded json. E.g
+
+```json
+"request_options": {
+            ...,
+            "content": {
+                            "content_type": "JSON",
+                            "json_mapping": {
+                                "nesting_delimiter": "__",
+                                "chunk_size": 1,
+                                "request_data_wrapper": "{ \"data\": [[data]]}",
```

Comment:
does [[data]] have to be always `[[data]]` ? Also use something else then `\"data\"` for clarity